### ISSUE TITLE
Warning: `support_autocorrect?` is deprecated. Use `cop.class.support_autocorrect?`

### ISSUE THREAD

**Issue by amomchilov** [rubocop/rubocop#13055]
## Expected behavior

No warnings

## Actual behavior

```
.../rubocop-1.65.0/lib/rubocop/cop/cop.rb:73: warning: `support_autocorrect?` is deprecated. Use `cop.class.support_autocorrect?`.
```

https://github.com/rubocop/rubocop/blob/28786cfef40d595c1aba29e1290d27e69b4c349c/lib/rubocop/cop/cop.rb#L73

## Steps to reproduce the problem

Run any cop that reports any offence.

## RuboCop version

I can't run `rubocop -V` right now, but it was version 1.65.

**Comment by amomchilov** [comment]
Ah this was already fixed in #13047, but it hasn't been released yet.

**Comment by koic** [comment]
Yes, this warning will disappear in the next release, but essentially, the cop's API needs to be updated to v1:
https://docs.rubocop.org/rubocop/1.65/v1_upgrade_notes.html

For instance, cop inheriting from `RuboCop::Cop::Cop` should be updated to inherit from `RuboCop::Cop::Base` instead. `RuboCop::Cop::Cop` is planned to be removed in the future, so please update to `RuboCop::Cop::Base`.
e.g., https://github.com/Shopify/rubocop-sorbet/pull/241
